weblate/glossary/forms.py

```diff
@@ -115,12 +95,12 @@ def clean(self):
     def as_kwargs(self):
         is_source = self.cleaned_data["translation"].is_source
         return {
-            "context": "",
+            "context": self.cleaned_data.get("context", ""),
             "source": self.cleaned_data["source"],
             "target": self.cleaned_data["source"]
             if is_source
             else self.cleaned_data.get("target"),
-            "auto_context": True,
+            "auto_context": bool(self.cleaned_data.get("auto_context", False)),
```

Comment:
As this now inherits from `NewUnitBaseForm`, might be better to call `super().as_kwargs()` and adjust changed fields only, something like (not tested):

```py
def as_kwargs(self):
    result = super().as_kwargs()
    if self.cleaned_data["translation"].is_source:
        result["target"] = result["source"]
    return result
````